pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -185,13 +187,18 @@ func buildAccessLogFromTelemetry(push *model.PushContext, spec *model.LoggingCon
 		var al *accesslog.AccessLog
 		switch prov := p.Provider.(type) {
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyFileAccessLog:
-			al = buildEnvoyFileAccessLogHelper(prov.EnvoyFileAccessLog)
+			// For default provider, fallback to Mesh Config for formatting options.
+			if p.Name == defaultEnvoyAccessLogProvider {
```

Comment:
we need update tests for this?

Reply:
The new test case that I have added, covers this